File: 796/rotate-string.py
class Solution:
    def rotateString(self, A, B):
        """
        :type A: str
        :type B: str
        :rtype: bool
        """
        if len(A) == 0 and len(B) == 0:
            return True

        def rotate(lst):
            lst.append(lst[0])
            return lst[1:]

        lst = list(A)
        for i in range(len(A)):
            lst = rotate(lst)
            if "".join(lst) == B:
                return True

        return False

File: 796/rotate-string.go
package main

import (
	"fmt"
	"strings"
)

func rotateString(A string, B string) bool {
	if len(A) == 0 && len(B) == 0 {
		return true
	}
	arr := make([]string, len(A))
	for i, ch := range A {
		arr[i] = string(ch)
	}
	for i := 0; i < len(A); i++ {
		arr = rotate(arr)
		if strings.Join(arr, "") == B {
			return true
		}
	}
	return false
}

func rotate(arr []string) []string {
	arr = append(arr, arr[0])
	return arr[1:]
}

func main() {
	A := "abcde"
	B := "cdeab"
	fmt.Println(rotateString(A, B))
}
